src/interface/src/app/plan/plan-map/plan-map.component.ts

```diff
@@ -57,6 +58,13 @@ export class PlanMapComponent implements AfterViewInit, OnDestroy {
       });
 
     setTimeout(() => this.map.invalidateSize(), 0);
+
+    this.planService.planState$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((state) => {
+        this.setCondition(state.mapConditionFilepath ?? '');
```

Comment:
can we check here if the mapConditionFilepath has changed in the state? because other things in the state could have changed instead.

Reply:
done! :D